packages/DApp/src/components/directory/DirectoryInfo.tsx

```diff
@@ -15,12 +15,51 @@ export function DirectoryInfo() {
   const { account } = useEthers()
   const { featuredVotingContract } = useContracts()
   const { getTypedFeatureVote } = useTypedFeatureVote()
-  const { activeVoting } = useFeaturedVotes()
   const { waku } = useWaku()
-
-  const featuredVotingState = getFeaturedVotingState(activeVoting)
+  const { activeVoting, votes } = useFeaturedVotes()
+  const featuredVotingState = useFeaturedVotingState(activeVoting)
   const castVotes = useContractFunction(featuredVotingContract, 'castVotes')
   const finalizeVoting = useContractFunction(featuredVotingContract, 'finalizeVoting')
+  const [loading, setLoading] = useState(false)
+
+  if (!activeVoting) {
+    return (
+      <InfoWrap>
+        <PageInfo
+          heading="Current directory"
+          text="Vote on your favourite communities being included in
+      Weekly Featured Communities"
+        />
+      </InfoWrap>
+    )
+  }
+
+  const evaluated = activeVoting.evaluated
+  const finalized = activeVoting.finalized
+  const votesCount: number = Object.values(votes).reduce(
+    (acc: number, curr: any) => acc + Object.keys(curr?.votes).length,
+    0
+  )
+
+  const beingFinalized = !evaluated && finalized
+  const beingEvaluated = evaluated && !finalized
+  const currentPosition = activeVoting.evaluatingPos
+  const firstFinalization = beingEvaluated && currentPosition === votesCount + 1
+
+  const votesLeftCount = votesCount - currentPosition + 1
+  const finalizeVotingLimit = firstFinalization
+    ? Math.min(votesCount, config.votesLimit)
+    : Math.min(votesLeftCount, config.votesLimit)
+  const batchCount = Math.ceil((beingFinalized ? votesCount + 1 : votesCount) / config.votesLimit)
+  const batchLeftCount = Math.ceil(votesLeftCount / config.votesLimit)
+
+  const batchDoneCount = batchCount - batchLeftCount
+
+  useEffect(() => {
+    if (finalizeVoting.state.status === 'Success' || castVotes.state.status === 'Success') {
+      history.go(0)
```

Comment:
please, could you explain me why are `history` statements necessary?

Reply:
When I get Success 'state' page is not re-rendered automatically, therefore we leverage this workaround to refresh page, so user can see current state.